lib/Installation/Partitioner/LibstorageNG/v4_3/ExpertPartitionerController.pm

```diff
@@ -223,4 +223,13 @@ sub edit_partition_on_gpt_disk {
     $self->get_edit_formatting_options_page()->press_next();
 }
 
+sub edit_partition_encrypt {
```

Comment:
We should be using common data structure. I added this property [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11446/files#diff-d0896615b84e6612e03efeebfb136807b5c5477174c46f006cf664177e4fbf2eR94) and [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11446/files#diff-b2319368889487698a0696ee6ceaecf9ede1d9ad60385b1d179e47fd1f3a038dR56) but now we might be able to do it all in 4.3, which is secondary goal. In other words, we should not be using `partitions_to_encrypt` it is kind of legacy debt we have introduced. Could you rethink it to follow common test data structure and optionally if it is not too complicated consider to do it all in 4.3?
Another point is that this should be 'edit_partition' and pass a parameter from test data to decide if we go with encryption or not. Otherwise we will repeat code when there will be no need for encryption.

Reply:
That would make sense if we want complex functions that cover all scenarios, but I thought the idea of using oop is to have simpler code. I can add the "encrypt" functionality in edit_partition_on_gtp_disk function and add "skip" option for all the mount finctions that don't include it. Then I can add " skip" for encryption in the test_data of the test that already use the function. WDYT?

Reply:
Instead of skip, the default behavior is not encrypted, so if we have in the test data that is encrypted we would run some extra method(s) in edit_partition_on_gpt_disk. If there is more than one make sense to wrap it in some function with `return unless defined $encrypted_device` or something like that. make sense?
The idea is too still keep it simple but not repeat the code for each kind of edition.

Reply:
That would be easier for me, but there is no consistency with the rest of the functions called by edit_partition_on_gpt_disk